internal/provider/atlas_migration_resource_test.go

```diff
@@ -319,17 +319,31 @@ func TestAccMigrationResource_RemoteDir(t *testing.T) {
 		dir   = migrate.MemDir{}
 		dbURL = fmt.Sprintf("sqlite://%s?_fk=true", filepath.Join(t.TempDir(), "sqlite.db"))
 		srv   = httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
-			var m struct {
-				Query     string `json:"query"`
-				Variables struct {
-					Input json.RawMessage `json:"input"`
-				} `json:"variables"`
-			}
+			type (
+				DeployRunContext struct {
+					Context struct {
+						TriggerType    string `json:"triggerType,omitempty"`
+						TriggerVersion string `json:"triggerVersion,omitempty"`
+					} `json:"context,omitempty"`
+				}
+				GraphQLQuery struct {
+					Query     string `json:"query"`
+					Variables struct {
+						Input json.RawMessage `json:"input"`
+					} `json:"variables"`
+				}
+			)
+			var m GraphQLQuery
 			require.NoError(t, json.NewDecoder(r.Body).Decode(&m))
 			switch {
 			case strings.Contains(m.Query, "query"):
 				writeDir(t, &dir, w)
 			case strings.Contains(m.Query, "reportMigration"):
+				var ctx DeployRunContext
```

Comment:
Use the structure from Atlas-Go-SDK
```suggestion
				var ctx atlas.DeployRunContext
```